### PR TITLE
Configure: Allow quoted values in VERSION

### LINKED ISSUE #11618: crypto/cversion.c compile error around OPENSSL_VERSION_TEXT
While building 3.0.0-alpha1 I noticed the following compilation error:

```
$ untar the 3.0.0-alpha1 tarball
$ ./config
$ make -j8
/Applications/Xcode.app/Contents/Developer/usr/bin/make depend && /Applications/Xcode.app/Contents/Developer/usr/bin/make _build_sw
cc  -Icrypto -I. -Iinclude -Iproviders/common/include -Iproviders/implementations/include -Icrypto/include  -DAES_ASM -DBSAES_ASM -DCMLL_ASM -DECP_NISTZ256_ASM -DGHASH_ASM -DKECCAK1600_ASM -DMD5_ASM -DOPENSSL_BN_ASM_GF2m -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DPOLY1305_ASM -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DVPAES_ASM -DWHIRLPOOL_ASM -DX25519_ASM -fPIC -arch x86_64 -O3 -Wall -DL_ENDIAN -DOPENSSL_BUILDING_OPENSSL -DOPENSSL_PIC -DOPENSSLDIR="\"/usr/local/ssl\"" -DENGINESDIR="\"/usr/local/lib/engines-3\"" -DMODULESDIR="\"/usr/local/lib/ossl-modules\"" -D_REENTRANT -DNDEBUG  -MMD -MF crypto/libcrypto-lib-cversion.d.tmp -MT crypto/libcrypto-lib-cversion.o -c -o crypto/libcrypto-lib-cversion.o crypto/cversion.c
crypto/cversion.c:50:16: error: expected ';' after return statement
        return OPENSSL_VERSION_TEXT;
               ^
include/openssl/opensslv.h:91:54: note: expanded from macro 'OPENSSL_VERSION_TEXT'
# define OPENSSL_VERSION_TEXT "OpenSSL 3.0.0-alpha1 "23 Apr 2020""
                                                     ^
1 error generated.
```

I could reproduce it by compiling 05feb0a0f1fecb6839888bb7590fb92be70d8d3c but not in the following d38997af165324b1ec0bbc0c6694daf7f503f080 commit. My suspicion is that the release date is wrongly quoted: https://github.com/openssl/openssl/blob/05feb0a0f1fecb6839888bb7590fb92be70d8d3c/VERSION#L6

Thanks,
Gregor

### PR DESCRIPTION
**Pull request by levitte**
Fixes #11618

### CODE REVIEW

File: Configure [hunk 1]
```diff
@@ -286,7 +286,14 @@ collect_information(
     qr/\s*(\w+)\s*=\s*(.*?)\s*$/ =>
         sub {
             # Only define it if there is a value at all
-            $version{uc $1} = $2 if $2 ne '';
+            if ($2 ne '') {
+                my $k = $1;
+                my $v = $2;
+                # Some values are quoted.  Trim the quotes
+                $v =~ s|^"||;
+                $v =~ s|"$||;
```
Comment (richsalz) [inline Configure:294]: I think this is more clear (and more correct, requiring balanced quotes)
```
$v = $1 if $v =~ /"(.*)"/;
```

Comment (levitte) [reply Configure:294]: Good point.

**Comment by levitte** [comment]
Hmm, somethings wrong with the fixup, hold on...

File: Configure (hunk 1 above)
Comment (richsalz) [reply Configure]: We are clearly in a Different World, if I'm making Perl suggestions to you. :)

Comment (levitte) [reply Configure]: Hahaha, I was thinking the same, like "Wait what?  Did he just....?"

**Comment by filimonov** [comment]
Checked, it fixes the issue 

fix can still fail if quotes or some special characters will appear in the middle of the version string, like 
```
RELEASE_DATE="23 Apr 2020 "now better" \ /*"
```
Maybe a bit safer to sanitize the string.

**Comment by levitte** [comment]
The answer is: don't do that. We won't.

**Comment by levitte** [comment]
Merged

6a960a94ccba41420c11ebd3eb82208b4681ee05 Configure: Allow quoted values in VERSION

**Pull request closed without merging**